src/lib/types.ts

```diff
@@ -0,0 +1,19 @@
+import type {
+	MapeoDoc,
+	MapeoValue,
+} from '@comapeo/schema' with { 'resolution-mode': 'import' }
+
+export type WriteableDocumentType = Extract<
+	MapeoDoc['schemaName'],
+	'field' | 'observation' | 'preset' | 'track' | 'remoteDetectionAlert'
+>
+export type WriteableValue<D extends WriteableDocumentType> = Extract<
+	MapeoValue,
+	{ schemaName: D }
+>
+export type WriteableDocument<D extends WriteableDocumentType> = Extract<
+	MapeoDoc,
+	{ schemaName: D }
+>
+
+export {}
```

Comment:
Curiosity question: What is this for and what does it do?

Reply:
It's so that the file gets treated as a module by TS. I don't know how much of a difference it makes since it's not super well documented but from what I've gathered, it's preferable to do this instead of having a declaration file when it comes to specifying types for your own code.

https://www.typescriptlang.org/docs/handbook/2/modules.html#non-modules

In some cases, I think it has consequences when it comes to either:

1. bundlers
2. TS resolving imports
3. TS emitting code

Again, not super sure - mostly added this out of intuition/habit 😄